ddl/ddl_worker.go

```diff
@@ -836,11 +836,12 @@ func (w *worker) HandleDDLJobTable(d *ddlCtx, job *model.Job) (int64, error) {
 	w.registerSync(job)
 
 	if runJobErr != nil && !dbterror.ErrPausedDDLJob.Equal(runJobErr) {
-		// Omit the ErrPausedDDLJob
-		w.jobLogger(job).Info("run DDL job failed, sleeps a while then retries it.",
-			zap.Duration("waitTime", GetWaitTimeWhenErrorOccurred()), zap.Error(runJobErr))
 		// In test and job is cancelling we can ignore the sleep.
-		if !(intest.InTest && job.IsCancelling()) {
+		// Or error is non-Retryable.
+		if !(intest.InTest && job.IsCancelling()) && errorIsRetryable(runJobErr, job) {
```

Comment:
Could we remove `job.IsCancelling()`?

Reply:
Done. This code also for speed up test before. https://github.com/pingcap/tidb/pull/41034